### PR TITLE
Enhance/marker position

### LINKED ISSUE #1626: 如何绘制对角线标注，并跟随图表的尺寸及数据变化

### LINKED ISSUE #1680: marker 定位能力增强，支持相对定位
- [x] markLine 支持 x1 y1
  - [x] markPoint 支持 x & y
  - [x] coordinate  支持回调
  - [x] positions 支持相对位置

### PR DESCRIPTION
**Pull request by kkxxkk2019**
marker 定位增强
  - [x] markLine 支持 x1 y1
  - [x] markPoint 支持 x & y
  - [x] coordinate  支持回调
  - [x] positions 支持相对位置

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [x] Enhancement feature
- [ ] Refactoring
- [ ] Update dependency
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Release
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #1680
close #1626 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

---

### 🚀 Summary

copilot:summary

### 🔍 Walkthrough

copilot:walkthrough

### CODE REVIEW

File: packages/vchart/src/component/marker/mark-point/interface/spec.ts [hunk 1]
```diff
@@ -1,34 +1,56 @@
...
 
 export type IMarkPointSpec = IMarkerSpec &
   (
+    | IMarkPointXYSpec
     | IMarkPointCoordinateSpec // 标注目标：数据元素
     | IMarkPointPositionsSpec
   ) &
   IMarkPointTheme<IMarkerSymbol>; // 标注目标：任意位置
 
+export type IMarkPointXYSpec = {
```
Comment (skie1997) [inline packages/vchart/src/component/marker/mark-point/interface/spec.ts:23]: 这个IMarkPointXYSpec的效果 不是可以用MarkerPositionPoint 或 IDataPointSpec配出来吗？

File: packages/vchart/src/component/marker/mark-point/interface/spec.ts [hunk 2]
```diff
@@ -1,34 +1,56 @@
...
   coordinates: IDataPointSpec;
   /**
-   * 被标注数据关联的series
+   * 对每个数据点转化后的画布坐标点进行偏移，该偏移值可以是像素值，也可以是 string 类型，如 '20%' 代表百分比
+   * 每个元素对应一个坐标点的偏移量
+   * @since 1.7.3
    */
-  relativeRelativeSeriesIndex?: number;
-  relativeRelativeSeriesId?: string;
+  coordinatesOffset?: OffsetPoint;
```
Comment (skie1997) [inline packages/vchart/src/component/marker/mark-point/interface/spec.ts:46]: 数据点的偏移是满足什么样的case呀？

File: packages/vchart/src/component/marker/mark-point/interface/spec.ts (hunk 1 above)
Comment (kkxxkk2019) [reply packages/vchart/src/component/marker/mark-point/interface/spec.ts:23]: 可以的，但是为了完备性考虑，还是要支持上，因为其他标注都有 x y

File: packages/vchart/src/component/marker/mark-point/interface/spec.ts (hunk 2 above)
Comment (kkxxkk2019) [reply packages/vchart/src/component/marker/mark-point/interface/spec.ts:46]: 这个主要来源编辑器场景，典型的是当数据点重叠时，需要调整下偏移

image

另外也有场景是期望对映射后的点做偏移，所以加了这个

**Review by skie1997 — DISMISSED** [review]

**Review by skie1997 — APPROVED** [review]

**Pull request merged**